sklearn/ensemble/iforest.py

```diff
@@ -206,7 +219,9 @@ def predict(self, X):
 
         depths += _average_path_length(n_samples_leaf)
 
-        if not isinstance(self.max_samples, (numbers.Integral, np.integer)):
```

Comment:
The problem is that the `base_estimator` (`ExtraTreeRegressor` in this case) does not know anything about `max_samples`. It is just given a set of samples to fit to by `_parallel_build_estimators()`